test/test_instance_requirements.rb

```diff
@@ -374,6 +377,22 @@
             instanciated = req.instanciate(plan)
             assert_equal srv, instanciated.model
         end
+
+        it "accepts a service selection expressed as an InstanceRequirements" do
+            srv_m  = Syskit::DataService.new_submodel
+            task_m = Syskit::TaskContext.new_submodel do
+                argument :key
+                provides srv_m, as: "test"
+            end
+
+            req = Syskit::InstanceRequirements.new([task_m])
+            srv = task_m.with_arguments(key: 10).test_srv
+            srv.name = "name" # closer to the original bug, which was a Definition
+            req_srv = req.select_service(srv)
```

Comment:
is srv `kind_of?(InstanceRequirements)`?

Reply:
as I see it is a srv_m

Reply:
It is. `task_m.test_srv` would be a `Models::BoundDataService`, but the `with_arguments` requires a `InstanceRequirements`

Generally speaking, both are interchangeable .... just not in this particular case.